internal/osbuild2/grub2_stage.go

```diff
@@ -83,3 +85,42 @@ func NewGrub2StageOptions(pt *disk.PartitionTable,
 
 	return &stageOptions
 }
+
+func NewGrub2StageOptionsUnified(pt *disk.PartitionTable,
+	kernelVer string,
+	uefi bool,
+	legacy string,
+	vendor string,
+	install bool) *GRUB2StageOptions {
+
+	rootFs := pt.FindMountable("/")
+	if rootFs == nil {
+		panic("root filesystem must be defined for grub2 stage, this is a programming error")
+	}
+
+	stageOptions := GRUB2StageOptions{
+		RootFilesystemUUID: uuid.MustParse(rootFs.GetFSSpec().UUID),
+		Legacy:             legacy,
+		WriteCmdLine:       common.BoolToPtr(false),
+	}
+
+	bootFs := pt.FindMountable("/boot")
+	if bootFs != nil {
+		bootFsUUID := uuid.MustParse(bootFs.GetFSSpec().UUID)
+		stageOptions.BootFilesystemUUID = &bootFsUUID
+	}
+
+	if uefi {
+		stageOptions.UEFI = &GRUB2UEFI{
+			Vendor:  vendor,
+			Install: install,
+			Unified: legacy == "",
```

Comment:
Why isn't this always `true`?

Reply:
We tried to minimise the changes in the manifest, since `UEFI` + `Legcay` is hybrid boot, which implies `unified` (same code path in osbuild), so in the case setting `Unified` is redundant.
Having said that, `Unified: true` is how we initially wrote the code and if we regenerate the manfifests anyway might as well always set it (even for the hybrid case).

Reply:
It should be.
Though it doesn't matter (functionally) in the cases where we use it, semantically it makes sense to have it pinned to true.  I just kept it that way to minimise the changes in the manifests and image info.

Reply:
ah, ok. just tripped me up due to the name of the function. a comment would solve it too

Reply:
I think it is either a comment or `Unified: True` and I am now swaying for the latter.

Reply:
Done